### PR TITLE
Adding omit-self

### PR DESCRIPTION
**Pull request by studioj**
Hi, 
I learned about your package on the python bytes podcast => https://pythonbytes.fm/episodes/show/208/dependencies-out-of-control-just-pip-chill.

They explained the use of the package but made a remark that it would be handy to not have "pip-chill" printed in the output but stated a workaround with **_grep_** for Linux or Mac users.

I, on the other hand, am a Windows user and don't have the convenient grep tool laying around.

I also have reworked your tests so they are more "cross-platformy"  and now can be run on the windows platform too
(I didn't run on Linux, but would guess it should work => the CI system will tell) (*edit: SUCCESS)

If I need to adjust something in the PR let me know, eager to help out. I noticed a few flake8 errors regarding line length in master itself dunno if I should bother too much?

### CODE REVIEW

**Comment by studioj** [comment]
I had to lock bandit to an earlier version since they dropped support for py2

File: tox.ini [hunk 1]
```diff
@@ -14,6 +14,6 @@ commands=flake8 pip_chill setup.py
 [testenv]
 setenv =
     PYTHONPATH = {toxinidir}:{toxinidir}/pip_chill
-deps = -rrequirements_dev.txt
+deps = -r requirements_dev.txt
```
Comment (studioj) [inline tox.ini:17]: this didnt change anything in the build result (I expected it to) 
I guess that space should be there though so I'll leave it

**Comment by studioj** [comment]
should i open this PR to the develop branch?

**Comment by rbanffy** [comment]
I ended up implementing a `--no-chill` cli option for removing pip-chill from the listing (wrote it after listening to the podcast, but pushed it today). Your other changes are, however, nice. Would you mind redoing this PR's other enhancements against the latest develop branch?

**Pull request closed without merging**